guest/filter/filter.go

```diff
@@ -46,11 +51,12 @@ var filter api.FilterPlugin
 //	}
 //
 // Note: If you need state, you can assign it with prefilter.SetPlugin.
-func SetPlugin(filterPlugin api.FilterPlugin) {
-	if filterPlugin == nil {
+func SetPlugin(pluginInitializer func(klog klogapi.Klog, jsonConfig []byte, h handleapi.Handle) api.FilterPlugin) {
```

Comment:
I think we should define the type of initializer function somewhere like the scheduling framework does.
And, let's name the type PluginFactory, following the framework.
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/runtime/registry.go#L31

Reply:
Thanks, fixed it